### PR TITLE
Add .cast, .retype across the package.

### PR DESCRIPTION
**Pull request by matanlurey**
And minimal testing, since we need DDC support for "real" testing.

### CODE REVIEW

File: lib/src/typed_wrappers.dart [hunk 1]
```diff
@@ -22,9 +22,7 @@ abstract class _TypeSafeIterableBase<E> implements Iterable<E> {
   bool any(bool test(E element)) => _base.any(_validate(test));
 
   // TODO: Dart 2.0 requires this method to be implemented.
```
Comment (srawlins) [inline lib/src/typed_wrappers.dart]: Nix this comment

File: lib/src/typed_wrappers.dart [hunk 2]
```diff
@@ -154,9 +149,7 @@ class TypeSafeList<E> extends TypeSafeIterable<E> implements DelegatingList<E> {
   Map<int, E> asMap() => new TypeSafeMap<int, E>(_listBase.asMap());
 
   // TODO: Dart 2.0 requires this method to be implemented.
```
Comment (srawlins) [inline lib/src/typed_wrappers.dart]: Nix this comment

File: lib/src/typed_wrappers.dart [hunk 3]
```diff
@@ -231,9 +224,7 @@ class TypeSafeList<E> extends TypeSafeIterable<E> implements DelegatingList<E> {
   }
 
   // TODO: Dart 2.0 requires this method to be implemented.
```
Comment (srawlins) [inline lib/src/typed_wrappers.dart]: Nix this comment

File: lib/src/typed_wrappers.dart [hunk 4]
```diff
@@ -314,9 +305,7 @@ class TypeSafeSet<E> extends TypeSafeIterable<E> implements DelegatingSet<E> {
...
```
Comment (srawlins) [inline lib/src/typed_wrappers.dart]: Nix this comment

File: lib/src/typed_wrappers.dart [hunk 5]
```diff
@@ -367,9 +356,7 @@ class TypeSafeQueue<E> extends TypeSafeIterable<E>
...
```
Comment (srawlins) [inline lib/src/typed_wrappers.dart]: Nix this comment

File: lib/src/typed_wrappers.dart [hunk 6]
```diff
@@ -367,9 +356,7 @@ class TypeSafeQueue<E> extends TypeSafeIterable<E>
...
-  Queue<T> retype<T>() {
-    throw new UnimplementedError('retype');
-  }
+  Queue<T> retype<T>() => new TypeSafeQueue<T>(_baseQueue.cast<T>());
```
Comment (srawlins) [inline lib/src/typed_wrappers.dart]: not `_baseQueue.retype<T>()`?

File: lib/src/typed_wrappers.dart [hunk 7]
```diff
@@ -443,9 +430,8 @@ class TypeSafeMap<K, V> implements DelegatingMap<K, V> {
   }
 
   // TODO: Dart 2.0 requires this method to be implemented.
```
Comment (srawlins) [inline lib/src/typed_wrappers.dart]: Nix this comment

File: lib/src/typed_wrappers.dart (hunk 6 above)
Comment (matanlurey) [reply lib/src/typed_wrappers.dart]: Oops, got it.

File: lib/src/typed_wrappers.dart (hunk 7 above)
Comment (matanlurey) [reply lib/src/typed_wrappers.dart]: Done.

File: lib/src/typed_wrappers.dart (hunk 5 above)
Comment (matanlurey) [reply lib/src/typed_wrappers.dart]: Done.

File: lib/src/typed_wrappers.dart (hunk 4 above)
Comment (matanlurey) [reply lib/src/typed_wrappers.dart]: Done.

File: lib/src/typed_wrappers.dart (hunk 3 above)
Comment (matanlurey) [reply lib/src/typed_wrappers.dart]: Done.

File: lib/src/typed_wrappers.dart (hunk 2 above)
Comment (matanlurey) [reply lib/src/typed_wrappers.dart]: Done.

**Pull request merged**